pipelines/smartseq2_single_sample/GroupQCOutputs.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Aug 21 11:07:46 2018
+
+@author: jishuxu
+"""
+
+from crimson import picard
+import argparse
+import os
+import pandas as pd
+
+def AggregatePicardMetricsRow(filenames, output_name):
+    """
+    pipeline output picard QC metrics at sinle cell/sample level.
+    This functin is called to merge/aggregate QC metrics by metrics type and
+    then merge multiple QC measurement into single matrix file.
+    In this file, column is sample/cell and row is QC metrics
+    :param files: metric files from pipeline outputs
+    :param met_name: metrics name with workflow name and subworkflow name
+    as prefix.such as 'run_pipelines.RunStarPipeline.alignment_summary_metrics'
+    """
+    # initial output
+    mets = {}
+    d = pd.DataFrame()
+    for file_name in filenames: 
+        cell_id = os.path.basename(file_name).split('_qc')[0]
+        mets[cell_id] = {}
+        parsed = picard.parse(file_name)
+        class_name = parsed['metrics']['class'].split('.')[2]
+        # Aignment metrics return multiple lines,
+        # but only output PAIRED-READS/third line
+        if class_name == "AlignmentSummaryMetrics":
+            # only parse out pair reads
+            met = parsed['metrics']['contents'][2]
```

Comment:
We can extract `parsed['metrics']['contents']` into a variable since it's used in several places throughout this function

Reply:
You could also split out the functionality that gets the metrics based on the class name into a separate function instead of having everything in one long function.